### ISSUE TITLE
Inspector panel performance degredates when kept opened

### ISSUE THREAD

**Issue by fcollonval** [jupyterlab/jupyterlab#11458]
I'll mention this here. I promise I'm not interested in moving goalposts, but maybe we can open another issue.

If I write this in jupyterlab, python kernel:  sns.relplot( | )  and place the cursor at the `|`, I get the inspector pane showing the docs. That's great.
When typing with the cursor there, we can notice that the kernel is working to update the inspector pane. I often notice some jank coming from keeping the inspector open when working, typing - then I'll have to close it, so that it doesn't spend time looking up stuff under the cursor all the time. This PR will probably already help the situation, less time spent on UI updates, but there could be more to gain (?)

Has anyone else seen this?

_Originally posted by @bluss in https://github.com/jupyterlab/jupyterlab/issues/11447#issuecomment-968136314_

**Comment by BodoBurger** [comment]
Yes, same for me, Jupyter Lab is lagging (>1 second after each keystroke) when the cursor is inside the brackets of a Pandas object and the "Contextual Help" is open.

Python 3.7.10
Jupyter 1.0.0
Juypterlab 3.2.4
Pandas 1.2.4

Windows 10 Pro
Firefox 94.0.1

**Comment by bluss** [comment]
Maybe the more precise title of the issue would be that jupyterlab UI performance is degraded when the inspector is open. It depends on where you're typing, of course, depends on the code how taxing it is to repeat the lookup.

**Comment by jolz** [comment]
I have performance issues whe dealing with a pandas DataFrames.

This issue seems to be similar to https://github.com/ipython-contrib/jupyter_contrib_nbextensions/issues/1275
Can the variable display be disabled in the Inspector?

**Comment by fcollonval** [comment]
> I have performance issues whe dealing with a pandas DataFrames.
> 
> This issue seems to be similar to [ipython-contrib/jupyter_contrib_nbextensions#1275](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/issues/1275) Can the variable display be disabled in the Inspector?

JupyterLab does not have the ability to display variable outside the debugger. This issue is about the inspector panel that will load the documentation for the current cursor position.

**Comment by jolz** [comment]
> > I have performance issues whe dealing with a pandas DataFrames.
> > This issue seems to be similar to [ipython-contrib/jupyter_contrib_nbextensions#1275](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/issues/1275) Can the variable display be disabled in the Inspector?
> 
> JupyterLab does not have the ability to display variable outside the debugger. This issue is about the inspector panel that will load the documentation for the current cursor position.

My example:
```python
import pandas as pd
import numpy as np
df = pd.DataFrame(np.random.randn(1000), index=pd.date_range("1/1/2000", periods=1000))
ts = pd.Series(np.random.randn(1000), index=pd.date_range("1/1/2000", periods=1000))
```
After execution clicking on `df` renders a help page like this: 
```

Type:        DataFrame
String form:
                   0
2000-01-01  0.975310
2000-01-02 -0.275562
2000-01-03 -0.097631
2000-01-04  1.517456
2000-01-05  0.865007
...              ...
2002-09-22  0.416387
2002-09-23 -0.235057
2002-09-24 -0.786755
2002-09-25 -1.284069
2002-09-26 -0.488189

[1000 rows x 1 columns]
Length:      1000
File:        /opt/conda/lib/python3.9/site-packages/pandas/core/frame.py
Source:     
class DataFrame(NDFrame, OpsMixin):
[...]
```

Navigation with cursor (with running kernel, with inspector open) let my cursor hang for a few seconds (`df`, `ts`). 
I assumed the cause in the "String form" in the inspector.
(i work with docker image "[jupyter/datascience notebook](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/selecting.html#jupyter-datascience-notebook7)") on a "not too powerful machine", so maybe it's just my ecosystem.

**Comment by fcollonval** [comment]
Thanks for providing an example @jolz This clarifies a lot the problem.

**Comment by KedirAdal** [comment]
I've also experienced the exact same issue working in pandas with a clean python environment and the latest jupyter + jupyterlab packages if the **Show Contextual Help** is opened.

- jupyter-client                   7.0.6   
- jupyter-core                     4.9.1   
- jupyter-packaging           0.11.1  
- jupyter-server                  1.11.2  
- jupyterlab                        3.3.0   
- jupyterlab-pygments       0.1.2   
- jupyterlab-server             2.10.3  

To add more on @jolz examples, once the dataframe _df_ is created, any actions after that such as clicking on or writing code within newly created cells freezes jupyterlab UI intermittently for a couple of seconds. The only workaround so far is to close the **Show Contextual Help**.
